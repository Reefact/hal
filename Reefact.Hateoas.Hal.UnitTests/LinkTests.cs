#region Usings declarations

using ApprovalTests;
using ApprovalTests.Reporters;

using Xunit;

#endregion

namespace Reefact.Hateoas.Hal.UnitTests {

    [UseReporter(typeof(BeyondCompareReporter))]
    public class LinkTests {

        [Fact]
        public void SelfLinkToStringTest() {
            Link selfLink = new("self") {
                Items = new LinkItemCollection {
                    new LinkItem("/orders")
                }
            };

            string hal = selfLink.ToString();

            Approvals.Verify(hal);
        }

        [Fact]
        public void CuriesLinkToStringTest() {
            Link curiesLink = new("curies") {
                Items = new LinkItemCollection(true) {
                    new LinkItem("http://example.com/docs/rels/{rel}") { Name = "ea", Templated = true }
                }
            };

            string hal = curiesLink.ToString();

            Approvals.Verify(hal);
        }

    }

}