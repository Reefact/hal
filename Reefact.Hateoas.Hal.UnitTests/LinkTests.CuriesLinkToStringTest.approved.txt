"curies": [
  {
    "name": "ea",
    "href": "http://example.com/docs/rels/{rel}",
    "templated": true
  }
]